openpaas-james/guice/encrypted-mailbox/src/main/java/com/linagora/openpaas/encrypted/MailboxConfiguration.java

```diff
@@ -0,0 +1,51 @@
+package com.linagora.openpaas.encrypted;
+
+import java.io.FileNotFoundException;
+import java.util.Objects;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.apache.james.utils.PropertiesProvider;
+
+public class MailboxConfiguration {
+    public static MailboxConfiguration parse(org.apache.james.server.core.configuration.Configuration configuration) throws ConfigurationException {
+        PropertiesProvider propertiesProvider = new PropertiesProvider(new FileSystemImpl(configuration.directories()),
+            configuration.configurationPath());
+
+        return parse(propertiesProvider);
+    }
+
+    public static MailboxConfiguration parse(PropertiesProvider propertiesProvider) throws ConfigurationException {
+        try {
+            Configuration configuration = propertiesProvider.getConfiguration("mailbox");
+            return new MailboxConfiguration(configuration.getBoolean("gpg.encryption.enable", false));
+        } catch (FileNotFoundException e) {
+            return new MailboxConfiguration(false);
+        }
+    }
+
+    private final boolean enableEncryption;
+
+    public MailboxConfiguration(boolean enableEncryption) {
+        this.enableEncryption = enableEncryption;
+    }
```

Comment:
Should we have one more constructor?
```java
 public MailboxConfiguration() {
        this.enableEncryption = true;
    }
```

Reply:
I don't think so. You think we might need a default constructor here?

Reply:
I think it's more convenient. At the `MailboxConfiguration.parse`, we also try to set with a default value (false).
```java
try {
            Configuration configuration = propertiesProvider.getConfiguration("mailbox");
            return new MailboxConfiguration(configuration.getBoolean("gpg.encryption.enable", false));
        } catch (FileNotFoundException e) {
            return new MailboxConfiguration(false);
        }
```

Reply:
That sounds "convenient" now because you only have one parameter in that new conf file, I'm not sure it would still make sense with more params.

This is our usual pattern regarding configuration files as well. If you change this, I guess a more global refactoring concerning configuration is required. Need to stay consistent in your implementations :)

But I honestly don't think this change is necessary IMO